tpm2/encoding.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2014, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpm2
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+func decodeCommandResponse(in []byte) (uint16, uint32, responseCode, error) {
+	var tag uint16
+	var size uint32
+	var status uint32
+
+	out := []interface{}{&tag, &size, &status}
+	err := unpack(in, out)
+	if err != nil {
+		return 0, 0, 0, fmt.Errorf("decoding response header: %v", err)
+	}
+
+	return tag, size, responseCode(status), nil
```

Comment:
I don't have the details of packing and unpacking in my head right now, but is it possible to allow unpack to unpack into a `responseCode`? If it's a matter of the type not being exported, then this would be a good argument for exporting it.